Sources/Brave/Frontend/Browser/BrowserViewController/BrowserViewController+WKNavigationDelegate.swift

```diff
@@ -349,6 +349,25 @@ extension BrowserViewController: WKNavigationDelegate {
     if ["http", "https", "data", "blob", "file"].contains(requestURL.scheme) {
       if navigationAction.targetFrame?.isMainFrame == true {
         tab?.updateUserAgent(webView, newURL: requestURL)
+        
+        let domain = Domain.getOrCreate(forUrl: requestURL, persistent: !isPrivateBrowsing)
+        
+        let shouldBlock = await AdBlockStats.shared.shouldBlock(
+          requestURL: requestURL, sourceURL: requestURL, resourceType: .document,
+          isAggressiveMode: domain.blockAdsAndTrackingLevel.isAggressive
+        )
+        
+        if shouldBlock {
+          var components = URLComponents(string: InternalURL.baseUrl)
+          components?.path = "/\(InternalURL.Path.blocked.rawValue)"
+          components?.queryItems = [URLQueryItem(name: "url", value: requestURL.absoluteString.escape()!)]
```

Comment:
Let's unwrap for safety. I do not see us not unwrapping escape of unescape functions in other places in code

Reply:
Fixed